google-cloud-examples/src/test/java/com/google/cloud/examples/storage/snippets/ITBucketSnippets.java

```diff
@@ -133,4 +139,50 @@ public void testBucket() throws InterruptedException {
     thrown.expect(StorageException.class);
     assertTrue(bucketSnippets.delete());
   }
+
+  @Test
+  public void testListBucketIamMembers() {
+    // Test an added Bucket-level IAM member is listed 
+    Policy policy = storage.getIamPolicy(BUCKET);
+    storage.setIamPolicy(BUCKET, policy.toBuilder().removeRole(StorageRoles.admin()).build());
+    policy = storage.getIamPolicy(BUCKET);
+    assertNull(policy.getBindings().get(StorageRoles.admin()));
+    storage.setIamPolicy(BUCKET, policy.toBuilder().addIdentity(StorageRoles.admin(),
+            Identity.user(USER_EMAIL)).build());
+    policy = storage.getIamPolicy(BUCKET);
+    assertTrue(policy.getBindings().get(StorageRoles.admin()).contains(Identity.user(USER_EMAIL)));
+    Policy snippetPolicy = bucketIamSnippets.listBucketIamMembers(BUCKET);
+    assertTrue(snippetPolicy.getBindings().get(StorageRoles.admin()).
+        contains(Identity.user(USER_EMAIL))); 
+  }
+
+  @Test
+  public void testAddBucketIamMemeber() {
+    // Test a member is added to Bucket-level IAM
+    Policy policy = storage.getIamPolicy(BUCKET);
```

Comment:
So rather than assuming you can get a fresh policy that is correct, actually return updatedPolicy from your snippet and review that.  (the reason is that there might be a hysteresis / some latency before the second request can get access to the data.